grafana/grafana
# CloudWatch: Deprecate dynamic labels feature toggle, remove support for Alias in frontend

**What is this feature?**
Similar to the [backend PR](https://github.com/grafana/grafana/pull/66494), in the frontend of the CloudWatch plugin, this PR:
* Removes any check for the Dynamic Label feature toggle, meaning Dynamic Label can no longer be disabled
   * setting the feature toggle off (cloudWatchDynamicLabels = false) will no longer have any effect
* Removes support for the Alias field to be used directly
* Still migrates Alias to Dynamic Label once: The migration will be handled on the fly, and will be propagated to the dashboard json model only when the user opens the query editor. Once the user has saved the dashboard model with the migrated label value, migration won't happen again. This also applies to Alert rules.

**Why do we need this feature?**

This was slated for removal in Grafana 10

**Which issue(s) does this PR fix?**:



Contributes to https://github.com/grafana/grafana/issues/66437

**Special notes for your reviewer:**
Alias has been replaced by Dynamic Label in https://github.com/grafana/grafana/issues/48434

Testing notes:
* Start on the main branch with the cloudWatchDynamicLabels toggle off
```
[feature_toggles]
cloudWatchDynamicLabels = false
```
* (Still on main branch) Create a dashboard with a CloudWatch datasource panel
* You should see the Alias field. Enter some former Alias pattern like `{{period}}`
* Save that dashboard.
* Now you can check out this PR and restart Grafana. 
* No matter if the feature toggle is false or true, you should see no longer see the Alias field. It should now be a Label field populated with a migrated pattern, e.g. ${PROP('Period')}
   * This Label will not be saved unless the dashboard is saved.  
   * Alert rules with Alias will not have their Label saved unless the rule is edited (opened and viewed) and saved again.

Please check that:
